app/src/main/java/org/oppia/android/app/topic/TopicFragmentPresenter.kt

```diff
@@ -70,6 +74,40 @@ class TopicFragmentPresenter @Inject constructor(
     return binding.root
   }
 
+  fun startSpotlight() {
+    viewModel.numberOfChaptersCompletedLiveData.observe(fragment) { numberOfChaptersCompleted ->
+      if (numberOfChaptersCompleted != -1) {
+        val lessonsTabView = tabLayout.getTabAt(computeTabPosition(TopicTab.LESSONS))?.view
+        lessonsTabView?.let { lessonsTabView ->
+          val lessonsTabSpotlightTarget = SpotlightTarget(
+            lessonsTabView,
+            "Find all your lessons here",
+            SpotlightShape.RoundedRectangle,
+            Spotlight.FeatureCase.TOPIC_LESSON_TAB
+          )
+          getSpotlightFragment().requestSpotlight(lessonsTabSpotlightTarget)
+
+          if (numberOfChaptersCompleted > 2) {
+            val revisionTabView = tabLayout.getTabAt(computeTabPosition(TopicTab.REVISION))?.view
+            val revisionTabSpotlightTarget = SpotlightTarget(
+              revisionTabView!!,
+              "Revise your lessons here",
+              SpotlightShape.RoundedRectangle,
+              Spotlight.FeatureCase.TOPIC_REVISION_TAB
+            )
+            getSpotlightFragment().requestSpotlight(revisionTabSpotlightTarget)
+          }
+        }
+      }
+    }
+  }
+
+  private fun getSpotlightFragment(): SpotlightFragment {
+    return activity.supportFragmentManager.findFragmentByTag(
+      SPOTLIGHT_FRAGMENT_TAG
+    ) as SpotlightFragment
```

Comment:
The returned fragment can be null--suggest adding a check here to make sure it's available when needed.

Reply:
i thought the fragment can never be null as we never actually exit out of the spotlight fragment. Adding a checkNotNull anyway.

Reply:
Technically the fragment can be absent in some situations which is why Kotlin can't make the guarantee here that it's never null. Based on how we've set stuff up _now_, it ought to never be null (hence why the clear crash is preferred). As the code changes, we want to catch failures that lead to a null fragment as early as possible.